### PR TITLE
storm-merger 1.5

### PR DESCRIPTION
**Pull request by tseemann**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by tseemann** [comment]
macos
```
         ^~~~~~~
         ^~~~~~~
1 error generated.
1 error generated.
1 error generated.
1 error generated.
make: *** [GeneralQueryDatasetFilter.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [DoubleKeyHashTable.o] Error 1
make: *** [Config.o] Error 1
make: *** [Alignment.o] Error 1
```

**Comment by tseemann** [comment]
```
uilding target: storm
Invoking: GCC C++ Linker
g++ -fopenmp  -o "storm"  ./Alignment.o ./Config.o ./DoubleKeyHashTable.o ./GeneralQueryDatasetFilter.o ./HashTable.o ./HashTableMethod.o ./OmegaGraphConstructor.o ./OmegaHashTable.o ./OverlapGraphConstructor.o ./PairEndReadMerger.o ./PairedEndRead.o ./PairedEndReadsMerger.o ./QueryDataset.o ./QueryDatasetFilter.o ./QueryRead.o ./SingleKeyHashTable.o ./SubjectDataset.o ./SubjectRead.o ./main.o   
Undefined symbols for architecture x86_64:
  "_omp_destroy_lock", referenced from:
      PairedEndRead::~PairedEndRead() in PairedEndRead.o
      QueryRead::~QueryRead() in QueryRead.o
  "_omp_get_wtime", referenced from:
      GeneralQueryDatasetFilter::start() in GeneralQueryDatasetFilter.o
      GeneralQueryDatasetFilter::printToFile() in GeneralQueryDatasetFilter.o
      OmegaGraphConstructor::start() in OmegaGraphConstructor.o
      OmegaGraphConstructor::printEdgesToFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in OmegaGraphConstructor.o
      OmegaGraphConstructor::start(bool) in OmegaGraphConstructor.o
      OmegaGraphConstructor::printEdgesToFileWithoutDuplicate(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in OmegaGraphConstructor.o
      OmegaGraphConstructor::printEdgesToFileWithoutTransitiveEdge(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in OmegaGraphConstructor.o
      ...
```

**Comment by tseemann** [comment]
Do i need to add `-lomp` for macos?

**Comment by sjackman** [comment]
On macOS either use `clang` with `libomp` or use GCC without `libomp`. I'd suggest removing this line: `depends_on "libomp" if OS.mac?` to use GCC without `libomp`.

**Comment by tseemann** [comment]
gcc without libomp is what i did first try - it didn't work, i then added libomp. still broken.

**Comment by tseemann** [comment]
```Invoking: GCC C++ Linker
g++ -fopenmp  -o "storm"  ./Alignment.o ./Config.o ./DoubleKeyHashTable.o ./GeneralQueryDatasetFilter.o ./HashTable.o ./HashTableMethod.o ./OmegaGraphConstructor.o ./OmegaHashTable.o ./OverlapGraphConstructor.o ./PairEndReadMerger.o ./PairedEndRead.o ./PairedEndReadsMerger.o ./QueryDataset.o ./QueryDatasetFilter.o ./QueryRead.o ./SingleKeyHashTable.o ./SubjectDataset.o ./SubjectRead.o ./main.o   
Undefined symbols for architecture x86_64:
  "_omp_destroy_lock", referenced from:
      PairedEndRead::~PairedEndRead() in PairedEndRead.o
      QueryRead::~QueryRead() in QueryRead.o
  "_omp_get_wtime", referenced from:
      GeneralQueryDatasetFilter::start() in GeneralQueryDatasetFilter.o
      GeneralQueryDatasetFilter::printToFile() in GeneralQueryDatasetFilter.o
      OmegaGraphConstructor::start() in OmegaGraphConstructor.o
      OmegaGraphConstructor::printEdgesToFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in OmegaGraphConstructor.o
      OmegaGraphConstructor::start(bool) in OmegaGraphConstructor.o
      OmegaGraphConstructor::printEdgesToFileWithoutDuplicate(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in OmegaGraphConstructor.o
      OmegaGraphConstructor::printEdgesToFileWithoutTransitiveEdge(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) in OmegaGraphConstructor.o
      ...
  "_omp_init_lock", referenced from:
      PairedEndRead::PairedEndRead() in PairedEndRead.o
      QueryRead::QueryRead() in QueryRead.o
```

File: Formula/storm-merger.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Scalable tool for merging PE reads with variable insert sizes"
+  homepage "https://bitbucket.org/yaoornl/align_test"
+  url "https://bitbucket.org/yaoornl/align_test/get/b514ea318f23.zip"
+  version "1.5"
+  sha256 "684e1dd6b95a0943cbaf769708386a6c3d3b2b5d792d65e404f8a43b4592c00d"
+
+  depends_on "gcc" # needs -fopenmp
+  depends_on "libomp" if OS.mac?
+
+  fails_with "clang" # needs -fopenmp
```
Comment (sjackman) [inline Formula/storm-merger.rb]:
```suggestion
  fails_with :clang # needs -fopenmp
```
Try that.

File: Formula/storm-merger.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class StormMerger < Formula
+  desc "Scalable tool for merging PE reads with variable insert sizes"
+  homepage "https://bitbucket.org/yaoornl/align_test"
+  url "https://bitbucket.org/yaoornl/align_test/get/b514ea318f23.zip"
+  version "1.5"
+  sha256 "684e1dd6b95a0943cbaf769708386a6c3d3b2b5d792d65e404f8a43b4592c00d"
+
+  depends_on "gcc" # needs -fopenmp
+  depends_on "libomp" if OS.mac?
```
Comment (sjackman) [inline Formula/storm-merger.rb]:
```suggestion
```

File: Formula/storm-merger.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class StormMerger < Formula
+  desc "Scalable tool for merging PE reads with variable insert sizes"
+  homepage "https://bitbucket.org/yaoornl/align_test"
+  url "https://bitbucket.org/yaoornl/align_test/get/b514ea318f23.zip"
+  version "1.5"
+  sha256 "684e1dd6b95a0943cbaf769708386a6c3d3b2b5d792d65e404f8a43b4592c00d"
+
+  depends_on "gcc" # needs -fopenmp
```
Comment (sjackman) [inline Formula/storm-merger.rb]:
```suggestion
  depends_on "gcc" if OS.mac? # needs -fopenmp
```

**Pull request closed without merging**